docs/modules/ROOT/pages/writing-upgradeable.adoc

```diff
@@ -0,0 +1,394 @@
+----
+
+Another difference between a `constructor` and a regular function is that Solidity takes care of automatically invoking the constructors of all ancestors of a contract. When writing an initializer, you need to take special care to manually call the initializers of all parent contracts:
+
+[source,solidity]
+----
+// contracts/MyContract.sol
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.0;
+
+import "@openzeppelin/upgrades/contracts/Initializable.sol";
+
+
+contract BaseContract is Initializable {
+    uint256 public y;
+
+    function initialize() public initializer {
+        y = 42;
+    }
+}
+
+
+contract MyContract is BaseContract {
+    uint256 public x;
+
+    function initialize(uint256 _x) public initializer {
+        BaseContract.initialize(); // Do not forget this call!
+        x = _x;
+    }
+}
+----
+
+[[use-upgradeable-packages]]
+=== Using Upgradeable Packages
+
+Keep in mind that this restriction affects not only your contracts, but also the contracts you import from a library. Consider for example https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v3.0.0/contracts/token/ERC20/ERC20.sol[`ERC20`] from OpenZeppelin Contracts: the contract initializes the token's name, symbol and decimals in its constructor.
+
+[source,solidity]
+----
+// @openzeppelin/contracts/token/ERC20/ERC20.sol
+pragma solidity ^0.6.0;
+
+  ...
+
+contract ERC20 is Context, IERC20 {
+
+  ...
+
+    string private _name;
+    string private _symbol;
+    uint8 private _decimals;
+
+    constructor (string memory name, string memory symbol) public {
+        _name = name;
+        _symbol = symbol;
+        _decimals = 18;
+    }
+
+  ...
+}
+----
+
+This means you should not be using these contracts in your OpenZeppelin Upgrades project. Instead, make sure to use `@openzeppelin/contracts-ethereum-package`, which is an official fork of OpenZeppelin Contracts that has been modified to use initializers instead of constructors. Take a look at how https://github.com/OpenZeppelin/openzeppelin-contracts-ethereum-package/blob/v3.0.0/contracts/token/ERC20/ERC20.sol[ERC20] looks in `@openzeppelin/contracts-ethereum-package`:
+
+[source,solidity]
+----
+// @openzeppelin/contracts-ethereum-package/contracts/token/ERC20/ERC20.sol
+pragma solidity ^0.6.0;
+  ...
+contract ERC20UpgradeSafe is Initializable, ContextUpgradeSafe, IERC20 {
+  ...
+    string private _name;
+    string private _symbol;
+    uint8 private _decimals;
+
+    function __ERC20_init(string memory name, string memory symbol) internal initializer {
+        __Context_init_unchained();
+        __ERC20_init_unchained(name, symbol);
+    }
+
+    function __ERC20_init_unchained(string memory name, string memory symbol) internal initializer {
+        _name = name;
+        _symbol = symbol;
+        _decimals = 18;
+    }
+  ...
+}
+----
+
+Whether using OpenZeppelin Contracts or another Ethereum Package, always make sure that the package is set up to handle upgradeable contracts.
```

Comment:
```suggestion
Whether using OpenZeppelin Contracts or another smart contract library, always make sure that the package is set up to handle upgradeable contracts.
```

Reply:
Resolved in: https://github.com/OpenZeppelin/openzeppelin-upgrades/pull/142/commits/2111130ed7681ce04989a5b15a87ed4173511454